cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/AbstractRuntimeTwillPreparer.java

```diff
@@ -139,10 +143,10 @@ abstract class AbstractRuntimeTwillPreparer implements TwillPreparer {
     this.twillSpec = twillSpec;
     this.programRunId = programRunId;
     this.programOptions = programOptions;
+    this.provisioner = SystemArguments.getProfileProvisioner(programOptions.getArguments().asMap());
     this.locationCache = locationCache;
     this.locationFactory = locationFactory;
     this.controllerFactory = controllerFactory;
-    this.extraOptions = cConf.get(io.cdap.cdap.common.conf.Constants.AppFabric.PROGRAM_JVM_OPTS);
```

Comment:
do you know why this was here to begin with? Does it mean we were double-setting these jvm options this whole time?

Reply:
Not sure why this was here to begin with. 
We started setting jvm options in DistributedProgramRunner as part of https://github.com/cdapio/cdap/pull/13226.

Reply:
> Does it mean we were double-setting these jvm options this whole time?

Yes, I think so

Reply:
ok I see